### ISSUE TITLE
clone fails if .$values are objects imported via reticulate

### ISSUE THREAD

**Issue by pfistfl** [mlr-org/paradox#276]
```r
library(paradox)
library(keras)
ParamSet$new(list(ParamUty$new("a")))    
ps = ParamSet$new(list(ParamUty$new("a")))
ps$values = list(a = optimizer_adam())
ps$clone(deep = TRUE)
```

> Error in py_get_attr_impl(x, name, silent) : 
  AttributeError: 'Adam' object has no attribute '.__enclos_env__'

Error most likely comes from Line 449 in https://github.com/mlr-org/paradox/blob/master/R/ParamSet.R

This currently breaks `mlr3keras` :)

**Comment by pfistfl** [comment]
The problem is, that `reticulate` overloads the `$` operator and therefore the access errors (and does not point to `NULL`).

**Comment by jakob-r** [comment]
Do we really want to to solve this in paradox? I think the go to solution is just construct the objects in the objective or in the trafo. 
My preferred solution would be to make the param a character and then just call `get(a, mode = "function")()` to construct the object. You are more flexible if you write expressions, though.

**Comment by pfistfl** [comment]
One problem here would be that then all extra args would also need to become parameters
i.e. `optimizer_adam()` has 8 args, which would then really blow up the param space.

**Comment by jakob-r** [comment]
What about
```r
ps$values = list(a = quote(optimizer_adam(foo = 1, bar = 2)))
```

**Comment by pfistfl** [comment]
Solved via #277
